mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs

```diff
@@ -1,161 +0,0 @@
-//
-// System.Reflection/CustomAttributeTypedArgument.cs
-//
-// Author:
-//   Zoltan Varga (vargaz@gmail.com)
-//   Carlos Alberto Cortez (calberto.cortez@gmail.com)
-//
-// Copyright (C) 2004 Novell, Inc (http://www.novell.com)
-//
-// Permission is hereby granted, free of charge, to any person obtaining
-// a copy of this software and associated documentation files (the
-// "Software"), to deal in the Software without restriction, including
-// without limitation the rights to use, copy, modify, merge, publish,
-// distribute, sublicense, and/or sell copies of the Software, and to
-// permit persons to whom the Software is furnished to do so, subject to
-// the following conditions:
-// 
-// The above copyright notice and this permission notice shall be
-// included in all copies or substantial portions of the Software.
-// 
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
-// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
-// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
-// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-//
-
-
-using System;
-using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Globalization;
-using System.Runtime.InteropServices;
-using System.Text;
-
-namespace System.Reflection {
-
-	[ComVisible (true)]
-	[Serializable]
-	public struct CustomAttributeTypedArgument {
-		Type argumentType;
-		object value;
-
-		public
-		CustomAttributeTypedArgument (Type argumentType, object value)
-		{
-			if (argumentType == null)
-				throw new ArgumentNullException ("argumentType");
-
-			this.argumentType = argumentType;
-			this.value = (value == null) ? null : CanonicalizeValue (value);
-
-			// MS seems to convert arrays into a ReadOnlyCollection
-			if (value is Array) {
```

Comment:
@akoeplinger As for [your comment](https://github.com/mono/mono/pull/10370#discussion_r213978561) about handling an "array case" in my prev PR - seems that it's still alive but they moved it to [here](https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/System.Private.Reflection.Core/src/System/Reflection/Runtime/CustomAttributes/RuntimeCustomAttributeData.cs#L189). Do I need to bring back this part as mono-conditional?

Reply:
Yes, and it seems the test failures confirm this: https://jenkins.mono-project.com/job/test-mono-pull-request-i386-osx/14257/testReport/junit/(root)/CustomAttributeDataTest/Arrays/

Btw. while you're at it please also reenable the test on mobile:
https://github.com/mono/mono/blob/3e841bf08911cd746bcf3473744706998fcc9bb7/mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs#L64-L66

The referenced bugzilla bug was marked fixed a few years ago: https://bugzilla.xamarin.com/show_bug.cgi?id=10263